mixer/adapter/kubernetes2/cache.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"errors"
+	"reflect"
+	"sync"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/mixer/pkg/adapter"
+)
+
+type (
+	// internal interface used to support testing
+	cacheController interface {
+		Run(<-chan struct{})
+		GetPod(string) (*v1.Pod, bool)
+		HasSynced() bool
+	}
+
+	controllerImpl struct {
+		clientset     kubernetes.Interface
```

Comment:
`clientset` is only used for initializing SharedInformer, seems unnecessary to store here.